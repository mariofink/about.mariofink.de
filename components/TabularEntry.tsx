interface props {
  term: string;
  children: React.ReactNode;
}

const TabularEntry = ({ term, children }: props) => (
  <div className="py-3 sm:grid sm:grid-cols-3 sm:gap-3 sm:px-3">
    <dt className="text-sm font-medium text-gray-500">{term}</dt>
    <dd className="mt-1 text-sm text-gray-900 sm:mt-0 sm:col-span-2">
      {children}
    </dd>
  </div>
);

export default TabularEntry;
