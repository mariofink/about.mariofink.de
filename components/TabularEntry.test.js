import React from "react";
import { render } from "@testing-library/react";
import TabularEntry from "./TabularEntry";

describe("TabularEntry", () => {
  it("renders", () => {
    const { container } = render(
      <TabularEntry term="My Term">My description</TabularEntry>
    );
    expect(container).toMatchSnapshot();
  });
});
