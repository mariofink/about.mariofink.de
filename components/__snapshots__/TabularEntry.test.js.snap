// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabularEntry renders 1`] = `
<div>
  <div
    class="py-3 sm:grid sm:grid-cols-3 sm:gap-3 sm:px-3"
  >
    <dt
      class="text-sm font-medium text-gray-500 dark:text-slate-400"
    >
      My Term
    </dt>
    <dd
      class="mt-1 text-sm text-gray-900 dark:text-slate-300 sm:mt-0 sm:col-span-2"
    >
      My description
    </dd>
  </div>
</div>
`;
